receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -155,6 +161,8 @@ func (jm *JobsMap) gc() {
 
 func (jm *JobsMap) maybeGC() {
 	// speculatively check if gc() is necessary, recheck once the structure is locked
+	jm.RLock()
```

Comment:
We could consider removing maybeGC() and calling 'go jm.gc()' directly. Either that or making jm.lastGC atomic.